### PR TITLE
Rackspace support WIP

### PR DESCRIPTION
**Pull request by GregSutcliffe**
Opening this for comments, as I don't think it's quite production ready yet. We don't get console output from this, but machine creation works, as does the Finish script. Requires an SSH to be present in the homedir of Foreman for the connection to work.

Comments?

### CODE REVIEW

File: lib/foreman/model/rackspace.rb [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      raise(ActiveRecord::RecordNotFound)
+    end
+
+    def create_vm args = { }
+      args = vm_instance_defaults.merge(args.to_hash.symbolize_keys)
+
+      x=vm_instance_defaults.merge(args.to_hash)
+      vm = client.servers.create x 
+      vm.wait_for { ready? }
+      vm.setup(:password => vm.password, :auth_methods=>["password"])
```
Comment (ohadlevy) [inline lib/foreman/model/rackspace.rb:34]: why password? arent we using key based login?

Comment (GregSutcliffe) [inline lib/foreman/model/rackspace.rb:34]: As I understand it, Rackspace isn't like EC2 - it returns a randomly generated password from the Fog call to create a new server. We need that to set up the SSH access. I'll check with Darksiez tomorrow though.

Comment (ohadlevy) [inline lib/foreman/model/rackspace.rb:34]: so you could actually login with that password instead of the key? in this case we wont need a key then.

Comment (darkskiez) [inline lib/foreman/model/rackspace.rb:34]: Yeah, but you only get the password in the creation call, and I wasn't sure where to store it,  so I made it install a ssh key instead.

**Comment by ohadlevy** [comment]
Then in this case, we don't need an ssh key at all.

**Comment by ohadlevy** [comment]
I'll try updating this one to RackSpace V2 API instead, any suggestions?

**Comment by GregSutcliffe** [comment]
I've never used the Rackspace v2 API, and I don't have access to the Rackspace accounts I was testing on. @darkskiez any thoughts?

**Comment by darkskiez** [comment]
I hope you can still create hosts on the v1 infrastructure.. the v2 servers give lower bang for your buck... (1/2 vcpu instead of 4, network throttling, etc)

Other than that I dont mind and can help test things.

**Comment by ohadlevy** [comment]
closes #263 added V2 support, reopen if you think we need V1.

**Pull request closed without merging**